### PR TITLE
Add KMplayer

### PR DESCRIPTION
**Pull request by Mailaender**
### Please confirm your submission meets all the criteria

- [X] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [X] My pull request follows the instructions at [App Submission][submission].
- [X] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [X] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [X] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** Please ping the KDE team.
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [X] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)* https://invent.kde.org/multimedia/kmplayer/-/commit/3def6507 https://invent.kde.org/multimedia/kmplayer/-/commit/7dafda63 
https://invent.kde.org/multimedia/kmplayer/-/commit/26bf80be https://invent.kde.org/multimedia/kmplayer/-/merge_requests/3

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by Mailaender** [comment]
bot, build org.kde.kmplayer

**Comment by hfiguiere** [comment]
cc @flathub/kde

**Comment by hfiguiere** [comment]
can't find the appdata.

**Comment by Mailaender** [comment]
bot, build org.kde.kmplayer

**Comment by hfiguiere** [comment]
```
org.kde.kmplayer:                  AppData problem: tag-invalid : <release> versions are not in order [0.11.5b before 0.12.0]
```

**Comment by Mailaender** [comment]
https://invent.kde.org/multimedia/kmplayer/-/merge_requests/3/diffs#00c3fbba0e29a092dc5dd993bbc8d82aaa2d32bf_349_354 Looks like the Flatpak parser is broken.

**Comment by hfiguiere** [comment]
not it's not. 

this is not sorted in order 
```
    <release version="0.11.5b" date="2016-07-10" />
    <release version="0.12.0" date="2016-06-02" />
    <release version="0.11.5a" date="2016-06-02" />
```

**Comment by Mailaender** [comment]
It was ordered by date.

**Comment by Mailaender** [comment]
bot, build org.kde.kmplayer

**Comment by hfiguiere** [comment]
> It was ordered by date.

and the validator was complaining about versions.

**Comment by Mailaender** [comment]
bot, build org.kde.kmplayer

**Comment by tsdgeos** [comment]
What's the plan for all the mplayer missing dependencies?

**Comment by Mailaender** [comment]
I added most of them now.

**Comment by Mailaender** [comment]
bot, build org.kde.kmplayer

**Comment by tsdgeos** [comment]
I honestly don't know enough about mplayer to say how important these are

  Disabled optional drivers:
    Input: vstream radio tv-v4l1 tv-dshow librtmp live555 nemesi cddb cdda smb 
    Codecs: crystalhd qtx win32 ilbc faac musepack libdca libgsm toolame twolame gif

**Comment by Mailaender** [comment]
I added giflib but it is not detected. toolame and twolame did not compile, those might need patches, but we already have MPEG2 decoders. musepack needs several patches. cddb is already in but is not detected. Not sure we need v4l1 when v4l2 is added via the runtime. Never heard the rest to be honest.

**Comment by barthalion** [comment]
mplayer isn't maintained for years now. Does it even make sense to add it to Flathub, when it's going to simply work worse than various mpv frontends or VLC?

**Comment by barthalion** [comment]
Huh!

File: org.kde.kmplayer.json [hunk 1]
```diff
@@ -0,0 +1,372 @@
...
+  "runtime-version": "5.15-21.08",
+  "sdk": "org.kde.Sdk",
+  "command": "kmplayer",
+  "rename-icon": "kmplayer",
+  "rename-desktop-file": "kmplayer.desktop",
+  "finish-args": [
+    "--share=ipc",
+    "--share=network",
+    "--socket=wayland",
+    "--socket=fallback-x11",
```
Comment (barthalion) [inline org.kde.kmplayer.json:13]:
```suggestion
    "--socket=x11",
```

It doesn't work on Wayland.

File: org.kde.kmplayer.json [hunk 2]
```diff
@@ -0,0 +1,372 @@
+{
+  "app-id": "org.kde.kmplayer",
+  "runtime": "org.kde.Platform",
+  "runtime-version": "5.15-21.08",
```
Comment (barthalion) [inline org.kde.kmplayer.json:4]:
```suggestion
  "runtime-version": "5.15-22.08",
```

Comment (travier) [reply org.kde.kmplayer.json:4]: Not yet, we're still testing this one.

**Pull request closed without merging**